tests/nlu/test_components.py

```diff
@@ -200,3 +200,17 @@ async def test_validate_requirements_raises_exception_on_component_without_name(
         await train(
             _config, data="./data/examples/rasa/demo-rasa.json", path=str(tmp_path),
         )
+
+
+async def test_validate_component_keys_raises_warning_on_invalid_key(tmp_path: Path,):
+    _config = RasaNLUModelConfig(
+        # config with a component that does not have a `name` property
```

Comment:
that was the part which confused me
```suggestion
        # config with a component that does not have a `confidence_threshold ` property
```

Reply:
ohhh lol my bad!

Reply:
I'll address the rest of the review later today